### PR TITLE
[add] Button component

### PR DESCRIPTION
**Pull request by calvintychan**
**This patch solves the following problem**
Add Button component to web support. #262 and #91 

- [x ] includes an interactive example

### CODE REVIEW

File: src/components/Button/index.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      accessibilityLabel,
+      color,
+      onPress,
+      title,
+      disabled
+    } = this.props;
+    const buttonStyles = [ styles.button ];
+    const textStyles = [ styles.text ];
+
+    if (color && Platform.OS === 'ios') {
```
Comment (necolas) [inline src/components/Button/index.js]: This will always be false on web

File: src/components/Button/index.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    } else if (color) {
+      buttonStyles.push({ backgroundColor: color });
+    }
+
+    if (disabled) {
+      buttonStyles.push(styles.buttonDisabled);
+      textStyles.push(styles.textDisabled);
+    }
+
+    invariant(
```
Comment (necolas) [inline src/components/Button/index.js]: We can drop this as the propTypes check is sufficient

File: src/components/Button/index.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      typeof title === 'string',
+      'The title prop of a Button must be a string',
+    );
+
+    const formattedTitle = Platform.OS === 'android' ? title.toUpperCase() : title;
```
Comment (necolas) [inline src/components/Button/index.js]: `OS` is always `web`

File: src/components/Button/index.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
+const ColorPropType = require('../../propTypes/ColorPropType');
```
Comment (necolas) [inline src/components/Button/index.js]: use `import`

File: src/components/Button/index.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+      'The title prop of a Button must be a string',
+    );
+
+    const formattedTitle = Platform.OS === 'android' ? title.toUpperCase() : title;
+
+    return (
+      <TouchableOpacity
+        accessibilityComponentType='button'
+        accessibilityLabel={accessibilityLabel}
+        accessibilityTraits={[ 'button' ]}
```
Comment (necolas) [inline src/components/Button/index.js]: Remove

File: src/components/Button/index.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    invariant(
+      typeof title === 'string',
+      'The title prop of a Button must be a string',
+    );
+
+    const formattedTitle = Platform.OS === 'android' ? title.toUpperCase() : title;
+
+    return (
+      <TouchableOpacity
+        accessibilityComponentType='button'
```
Comment (necolas) [inline src/components/Button/index.js]: use `accessibilityRole`

File: src/components/Button/index.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        accessibilityLabel={accessibilityLabel}
+        accessibilityTraits={[ 'button' ]}
+        disabled={disabled}
+        onPress={onPress}>
+        <View style={buttonStyles}>
```
Comment (necolas) [inline src/components/Button/index.js]: Is this nested View needed? We can apply the styles to `TouchableOpacity`

**Comment by calvintychan** [comment]
Got it. I will make the changes. Thanks for the input!

File: src/components/Button/index.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
+import ColorPropType from '../../propTypes/ColorPropType';
+import React, { Component, PropTypes } from 'react';
+import StyleSheet from '../../apis/StyleSheet';
+import TouchableOpacity from '../Touchable/TouchableOpacity';
+import Text from '../Text';
+
+class Button extends Component {
+
+  static propTypes = {
+    /**
```
Comment (necolas) [inline src/components/Button/index.js]: Add `accessibilityLabel`

File: src/components/Button/index.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
+import ColorPropType from '../../propTypes/ColorPropType';
+import React, { Component, PropTypes } from 'react';
+import StyleSheet from '../../apis/StyleSheet';
+import TouchableOpacity from '../Touchable/TouchableOpacity';
+import Text from '../Text';
+
+class Button extends Component {
+
```
Comment (necolas) [inline src/components/Button/index.js]: Remove blank line

**Comment by necolas** [comment]
Thanks, I made a few additional changes and committed in f8d5c154052f9a7c56cb863d00fc1e4eb806ce29

**Comment by walidvb** [comment]
Was this ever merged? 🤔

**Pull request closed without merging**